ibis/expr/datatypes/value.py

```diff
@@ -266,7 +272,19 @@ def normalize(typ, value):
             return float(value)
         except ValueError:
             raise TypeError("Unable to normalize {value!r} to {dtype!r}")
-    elif dtype.is_string() and not dtype.is_json():
+    elif dtype.is_json():
+        if isinstance(value, str):
+            try:
+                json.loads(value)
+            except json.JSONDecodeError:
+                raise TypeError(f"Invalid JSON string: {value!r}")
+            else:
+                return value
+        else:
+            return json.dumps(value)
```

Comment:
What's the rationale for the logic of "decode if it's a string, otherwise encode"?

Reply:
This ensures that the input string is a valid json string as well as an input object is json encodable.